src/mds/CInode.cc

```diff
@@ -5383,162 +5398,90 @@ void CInode::queue_export_pin(mds_rank_t target)
       // clear aux subtrees ?
       queue = dir->state_test(CDir::STATE_AUXSUBTREE);
     }
-    if (queue) {
-      state_set(CInode::STATE_QUEUEDEXPORTPIN);
-      mdcache->export_pin_queue.insert(this);
+
+    if (queue)
       break;
-    }
+  }
+  if (queue) {
+    state_set(CInode::STATE_QUEUEDEXPORTPIN);
+    mdcache->export_pin_queue.insert(this);
   }
 }
 
 void CInode::maybe_export_pin(bool update)
 {
   if (!g_conf()->mds_bal_export_pin)
     return;
-  if (!is_dir() || !is_normal())
+  if (!is_dir())
```

Comment:
why remove this check?

Reply:
because '!is_dir() || !is_normal()' is always true

Reply:
> because '!is_dir() || !is_normal()' is always true

uh? No it's not? The main point of that check was to exclude root.

Reply:
you are right